src/Symfony/Bundle/FrameworkBundle/Templating/Loader/TemplateLocator.php

```diff
@@ -72,13 +77,29 @@ public function locate($template, $currentPath = null, $first = true)
         $key = $this->getCacheKey($template);
 
         if (isset($this->cache[$key])) {
-            return $this->cache[$key];
+            return '.' === $this->cache[$key][0] ? $this->rootDir.$this->cache[$key] : $this->cache[$key];
         }
 
         try {
-            return $this->cache[$key] = $this->locator->locate($template->getPath(), $currentPath);
+            $path = $this->locator->locate($template->getPath(), $currentPath);
+
+            if ($this->rootDir && 0 === strpos($path, $this->rootDir)) {
+                return $this->rootDir.($this->cache[$key] = './'.substr($path, strlen($this->rootDir)));
```

Comment:
Tip: As the length of the rootDir doesn't change at runtime you can cache it as a property or static local variable.